ui/src/pages/HomePage/Register/index.js

```diff
@@ -1,60 +1,167 @@
-            href="https://www.ht2labs.com/learning-locker/register-installation/#register"
-            target="_blank"
-            onClick={setRegistered}
-            rel="noopener noreferrer">
-            Register
-          </RegistrationLink>
+    if (SITE_SETTINGS_OS_REG_CODE === codeRef.current.value) {
+      setModalOpen(false);
+      setRegistered();
+    } else {
+      setError('Incorrect code, please try again.');
+    }
+  };
 
-          <DontShowAgainText
-            className="pull-left"
-            onClick={setRegistered}>
-            Don&#39;t show again
-          </DontShowAgainText>
+  const renderInviteModal = () => {
+    const codeId = uuid.v4();
 
-        </div>
-      </div>
-    </CardContainer>
-  </Card>);
+    return (
+      <Portal isOpened={modalOpen}>
+        <span>
+          <div className="modal animated fast fadeIn">
+            <div className="modal-dialog">
+              <div className="modal-content">
+                <div className="modal-header modal-header-bg">
+                  <button type="button" className="close" aria-label="Close" onClick={() => setModalOpen(false)}><span aria-hidden="true">&times;</span></button>
+                  <h4 className="modal-title">Enter Code</h4>
+                </div>
+                <form onSubmit={onSubmit}>
+                  <div className="modal-body clearfix">
+                    <div
+                      className={classNames({
+                        'form-group': true,
+                        'has-error': error
+                      })} >
+                      <label htmlFor={codeId}>Registration Code</label>
+                      <input
+                        autoFocus
+                        id={codeId}
+                        className="form-control"
+                        type="text"
+                        ref={codeRef} />
+                      {error &&
+                        (<span className="help-block">
+                          <ValidationList errors={fromJS([error])} />
+                        </span>
+                        )
+                      }
+                    </div>
+                  </div>
+                  <div className="modal-footer">
+                    <div className="col-xs-12 clearfix">
+                      <button className="btn btn-primary btn-sm pull-right">
+                        Confirm
+                      </button>
+                    </div>
+                  </div>
+                </form>
+              </div>
+            </div>
+            <div className="modal-backdrop" onClick={() => setModalOpen(false)} />
+          </div>
+        </span>
+      </Portal>
+    );
+  };
+
+  const setProceedOnce = () => {
+    sessionStorage.setItem('proceedOnce', true);
+    document.dispatchEvent(new Event('setProceedOnce'));
+  };
+
+  return (<div>
+    { renderInviteModal() }
+    <div className="col-md-7" style={{ marginTop: 10 }}>
```

Comment:
@lucuilearningpool , instead of giving `marginTop` to container's children, you can add padding to the parent container